spec/patterns/firewalls_spec.rb

```diff
@@ -2,109 +2,593 @@
+                                 "protocol"=>"TCP", "action"=>"Dropping",
+                                 "reason"=>"MSS exceeded, MSS 1380, data 1460"
+    end
+  end
+
+end
+
+describe_pattern "CISCOFW419002", ['legacy', 'ecs-v1'] do
+
+  let(:message) do
+    "%ASA-4-419002: Duplicate TCP SYN from OUTSIDE:10.10.66.2/65087 to INSIDE:10.10.1.6/443 with different initial sequence number."
+  end
+
+  include_examples 'top-level namespaces', CISCOFW_ALLOWED_TOP_LEVEL_NAMESPACES, if: -> { ecs_compatibility? }
+
+  it 'matches' do
+    if ecs_compatibility?
+      expect(subject).to include "event"=>{"reason"=>"Duplicate TCP SYN"},
+                                 "source"=>{"port"=>65087, "ip"=>"10.10.66.2"},
+                                 "destination"=>{"port"=>443, "ip"=>"10.10.1.6"},
+                                 "observer"=>{"egress"=>{"interface"=>{"name"=>"INSIDE"}}, "ingress"=>{"interface"=>{"name"=>"OUTSIDE"}}}
+    else
+      expect(subject).to include "src_ip"=>"10.10.66.2", "src_port"=>"65087", "src_interface"=>"OUTSIDE",
+                                 "dst_ip"=>"10.10.1.6", "dst_port"=>"443", "dst_interface"=>"INSIDE",
+                                 "reason"=>"Duplicate TCP SYN"
+    end
+  end
+
+end
+
+describe_pattern "CISCOFW602303_602304", ['legacy', 'ecs-v1'] do
+
+  let(:message) do
+    "%ASA-6-602303: IPSEC: An outbound LAN-to-LAN SA (SPI= 0xF81283) between 91.240.17.178 and 192.168.2.2 (user= admin) has been created."
+  end
+
+  include_examples 'top-level namespaces', CISCOFW_ALLOWED_TOP_LEVEL_NAMESPACES, if: -> { ecs_compatibility? }
+
+  it 'matches' do
+    if ecs_compatibility?
+      expect(subject).to include("cisco"=>{"asa"=>{
+                                    "network"=>{"direction"=>"outbound", "type"=>"IPSEC"},
+                                    "outcome"=>"created",
+                                    "ipsec"=>{"spi"=>"0xF81283", "tunnel_type"=>"LAN-to-LAN"}}},
+                                 "destination"=>{"ip"=>"192.168.2.2"},
+                                 "source"=>{"ip"=>"91.240.17.178", "user"=>{"name"=>"admin"}})
+    else
+      expect(subject).to include "protocol"=>"IPSEC", "direction"=>"outbound", "tunnel_type"=>"LAN-to-LAN",
+                                 "src_ip"=>"91.240.17.178", "dst_ip"=>"192.168.2.2",
+                                 "spi"=>"0xF81283", "user"=>"admin", "action"=>"created"
+    end
+  end
+
+end
+
+describe_pattern "CISCOFW710001_710002_710003_710005_710006", ['legacy', 'ecs-v1'] do
+
+  let(:message) do
+    "%PIX-7-710001: TCP access requested from 192.168.1.2/2354 to inside:192.168.1.1/443"
+  end
+
+  include_examples 'top-level namespaces', CISCOFW_ALLOWED_TOP_LEVEL_NAMESPACES, if: -> { ecs_compatibility? }
+
+  it 'matches' do
+    if ecs_compatibility?
+      expect(subject).to include("destination"=>{"port"=>443, "ip"=>"192.168.1.1"},
```

Comment:
I would expect to see `source` validated here, since we validate its counterpart when ECS is disabled.